src/core/reactor.ts

```diff
@@ -983,25 +983,57 @@ protected _getFirstReactionOrMutation(): Reaction<any> | undefined {
             }
 
             // Add the new edge.
+            // Add edge when we are able to add graph
+            // if(isAbleToAdd(dst, src)) graph.addEdge(dst, src)
             graph.addEdge(dst, src)
 
             // 1) check for loops
+            let hasCycle = false
             if (graph.hasCycle()) {
-                return false
+                hasCycle = true
+                // throw Error("New connection introduces cycle.")
             }
 
             // 2) check for direct feed through.
+            let hasDirectFeedThrough = false
             let inputs = this._findOwnInputs()
-            for (let output of this._findOwnOutputs()) {
+            let outputs = this._findOwnOutputs()
+
+
+            for (let output of outputs) {
                 let newReachable = graph.reachableOrigins(output, inputs)
                 let oldReachable = this._causalityGraph.reachableOrigins(output, inputs)
 
+                // Print newReachable and oldReachable origin
+                for (let new_ori of newReachable) {
+                    if (new_ori instanceof Port) {
+                        console.log("New Origin: ", new_ori._getFullyQualifiedName())
+                    }              
+                }
+                for (let old_ori of oldReachable) {
+                    if (old_ori instanceof Port) {
+                        console.log("Old Origin: ", old_ori._getFullyQualifiedName())
+                    }              
+                }
+                console.log("!!!")    
```

Comment:
Nit: should be deleted?

Reply:
It was reflected in https://github.com/lf-lang/reactor-ts/pull/108/commits/9019da3204bf0e1773451d2c1a27b0485e3ac8e1. This is my fault.